src/shogun/regression/GLM.cpp

```diff
@@ -7,18 +7,32 @@
 #include <utility>
 using namespace shogun;
 
-GLM::GLM(): LinearMachine(){
-    init();
+GLM::GLM() : LinearMachine()
+{
+	init();
 }
 
-void GLM::init(){
-
+void GLM::init()
+{
+	SG_ADD(&m_alpha, "alpha", "alpha parameter", ParameterProperties::HYPER);
+	SG_ADD(&m_lambda, "lambda", "lambda parameter", ParameterProperties::HYPER);
+	SG_ADD(
+	    (std::shared_ptr<SGObject>*)&m_descend_updater, "descend_updater",
+	    "Descend Updater used for updating weights",
+	    ParameterProperties::SETTING);
```

Comment:
as for those lines, I think that's what causing the build to fail

Reply:
any thoughts? @karlnapf

Reply:
Not sure what you mean

Reply:
I get an error when I test as you could tell in the CI, in the SGObject test, so I have a feeling this is what causing it. what do you think?

Reply:
when you report such thing, could you please make sure that you copy here the link to the CI line where you think the error is....

Reply:
The issue is probably that m_descend_updater (`DescendUpdater`) is never initialised, so you are serialising a nullptr, and I think that might be causing issues... @vigsterkr ?

Reply:
Hence https://dev.azure.com/shogunml/shogun/_build/results?buildId=3277&view=logs&j=c857963e-3b4d-5192-8776-ae81779de3af&t=542541c1-2650-5e44-f82f-a76d8c4f5255&l=216

Reply:
mmm but this should catch that:
https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/io/serialization/JsonSerializer.cpp#L178